### DISCUSSION TITLE
Customize the configuration of CodeMirror in `text` mode

### DISCUSSION THREAD

**Discussion by topliceanu** [josdejong/svelte-jsoneditor#368] [Ideas]
Hi 👋 
I'm looking for something similar to the solution in this issue https://github.com/josdejong/jsoneditor/issues/1082
Thanks!

**Comment by josdejong** [comment]
Thanks, that is a good suggestion. It would be helpful if you can adjust configuration of the CodeMirror editor (used for `text` mode) yourself.

I think we should introduce a callback `onCreateCodeMirror`, which allows you to adjust configuration of the created code mirror instance, and allows you to simply return a different instance if that is handier.

```ts
onCreateCodeMirror: (codeMirrorView: EditorView) => EditorView | undefined
```

It would be good to also create an example in the `/src/routes/examples` section demonstrating how to change some of the configuration, like line wrapping. Anyone able to help implementing this callback? Help would be welcome.

As for your actual case: the extension doing the text wrapping is located here: 

https://github.com/josdejong/svelte-jsoneditor/blob/af6f5f2f69414b5018050ac9f97b90dadac2676a/src/lib/components/modes/textmode/TextMode.svelte#L492

**Comment by josdejong** [comment]
Moving this idea to the [Discussions > Ideas](https://github.com/josdejong/svelte-jsoneditor/discussions/categories/ideas) section since we are not actively working on it. Help implementing this would be very welcome.
